Rivendell was intended to be installed with sox-12 on the side. This patch replaces deprecated 
so that we might use sox-14 instead. The -w is now -2, -s has been removed (it requested the 
the default resampling algorithm), volume adjustment syntax has changed, and resampling has 
been simplifed. 
diff -urN rivendell-1.2.2/scripts/rd_export_file rivendell-1.2.2.gentoo/scripts/rd_export_file
--- rivendell-1.2.2/scripts/rd_export_file	2008-09-18 14:02:16.000000000 -0500
+++ rivendell-1.2.2.gentoo/scripts/rd_export_file	2009-03-18 11:11:03.000000000 -0500
@@ -79,13 +79,6 @@
 CUSTOM_CMD=${13}
 
 #
-# Resampling Flag
-#
-if [ $SAMPRATE_IN != $SAMPRATE_OUT ] ; then
-  RESAMPLE_FLAG="resample -q"
-fi
-
-#
 # MPEG Stuff
 #
 if [ $CHANS_OUT = 1 ] ; then
@@ -125,18 +118,18 @@
 # Get Peak Level
 #
 if [ $NORMAL_LEVEL != 0 ] ; then
-  PEAK_LEVEL=`sox $WORK -e stat -v 2>&1 | grep -v ^sox` 
+  PEAK_LEVEL=`sox $WORK -n stat -v 2>&1 | grep -v ^sox` 
   SCALE=`echo "$NORMAL_LEVEL * $PEAK_LEVEL" | bc -l`
   if [ "$SCALE" == "1.00000000000000000000" ]; then
     SCALE="1"
   fi
-  SOX_SCALE="-v $SCALE"
+  SOX_SCALE="vol $SCALE"
   chmod 644 $WORK
 fi
 
 case "$FORMAT_OUT" in
   0)  # PCM16
-    sox $SOX_SCALE $WORK -r $SAMPRATE_OUT -c $CHANS_OUT "$FILE_OUT" $RESAMPLE_FLAG
+    sox $WORK -r $SAMPRATE_OUT -c $CHANS_OUT "$FILE_OUT" $SOX_SCALE
   ;;
 
   1)  # MPEG Layer 1
@@ -144,19 +137,19 @@
   ;;
 
   2)  # Mpeg Layer 2
-    sox $SOX_SCALE $WORK -r $SAMPRATE_OUT -c $CHANS_OUT -t wav - $RESAMPLE_FLAG | toolame -t 0 -s $TOOLAME_SAMPRATE -m $MPEG_MODE -b $BITRATE_OUT /dev/stdin "$FILE_OUT"
+    sox $WORK -r $SAMPRATE_OUT -c $CHANS_OUT -t wav - $SOX_SCALE | toolame -t 0 -s $TOOLAME_SAMPRATE -m $MPEG_MODE -b $BITRATE_OUT /dev/stdin "$FILE_OUT"
   ;;
 
   3)  # MPEG Layer 3
-    sox $SOX_SCALE $WORK -r $SAMPRATE_OUT -c $CHANS_OUT -t raw - $RESAMPLE_FLAG | lame --silent -r -x -s $SAMPRATE_OUT -m $MPEG_MODE $LAME_RATE - "$FILE_OUT"
+    sox $WORK -r $SAMPRATE_OUT -c $CHANS_OUT -t raw - $SOX_SCALE | lame --silent -r -x -s $SAMPRATE_OUT -m $MPEG_MODE $LAME_RATE - "$FILE_OUT"
   ;;
 
   4)  # FLAC
-    sox $SOX_SCALE $WORK -r $SAMPRATE_OUT -c $CHANS_OUT -t raw - $RESAMPLE_FLAG | flac -f --silent --best --endian=little --sign=signed --bps=16 --channels=$CHANS_OUT --sample-rate=$SAMPRATE_OUT --force-raw-format -o "$FILE_OUT" -
+    sox $WORK -r $SAMPRATE_OUT -c $CHANS_OUT -t raw - $SOX_SCALE | flac -f --silent --best --endian=little --sign=signed --bps=16 --channels=$CHANS_OUT --sample-rate=$SAMPRATE_OUT --force-raw-format -o "$FILE_OUT" -
   ;;
 
   5)  # OggVorbis
-    sox $SOX_SCALE $WORK -r $SAMPRATE_OUT -c $CHANS_OUT -t raw - $RESAMPLE_FLAG | oggenc -r -B 16 -C $CHANS_OUT -R $SAMPRATE_OUT -q $QUALITY_OUT -o "$FILE_OUT" -
+    sox $WORK -r $SAMPRATE_OUT -c $CHANS_OUT -t raw - $SOX_SCALE | oggenc -r -B 16 -C $CHANS_OUT -R $SAMPRATE_OUT -q $QUALITY_OUT -o "$FILE_OUT" -
   ;;
 
   *)  # Custom Format
@@ -165,7 +158,7 @@
       rm -f $PEAK
       exit 1
     fi
-    sox $SOX_SCALE $WORK -r $SAMPRATE_OUT -c $CHANS_OUT -t raw - $RESAMPLE_FLAG | $CUSTOM_CMD
+    sox $WORK -r $SAMPRATE_OUT -c $CHANS_OUT -t raw - $SOX_SCALE | $CUSTOM_CMD
   ;;
 esac
 
diff -urN rivendell-1.2.2/scripts/rd_import_file rivendell-1.2.2.gentoo/scripts/rd_import_file
--- rivendell-1.2.2/scripts/rd_import_file	2007-02-14 15:59:12.000000000 -0600
+++ rivendell-1.2.2.gentoo/scripts/rd_import_file	2009-03-18 11:05:33.000000000 -0500
@@ -165,31 +165,20 @@
 case "$FORMAT_OUT" in
     0)
     function masterize() {
-	SOX_OPTIONS="$1 -t raw -s -w -c $CHANS_OUT -r $SAMPRATE_OUT -"
+	SOX_OPTIONS="$1 -2 -c $CHANS_OUT -r $SAMPRATE_OUT -t raw -"
 
-	if [ $SAMPRATE_OUT -ne $SAMPRATE_IN ]; then
-#	    SOX_OPTIONS="$SOX_OPTIONS resample -q"
-	    SOX_OPTIONS="$SOX_OPTIONS resample"
-	fi
-    
 	if [ $3 != 1  ]; then
-	    SOX_OPTIONS="-v $3 $SOX_OPTIONS"
+	    SOX_OPTIONS="$SOX_OPTIONS vol $3"
     fi
     sox $SOX_OPTIONS | rdfilewrite --channels=$4 --sample-rate=$5 $2
     }
     ;;
     1)
     function masterize() {
-	SOX_OPTIONS="$1 -w -c $CHANS_OUT -r $SAMPRATE_OUT -t raw"
+	SOX_OPTIONS="$1 -2 -c $CHANS_OUT -r $SAMPRATE_OUT -t raw -"
 	if [ $3 != 1  ]; then
-	    SOX_OPTIONS="-v $3 $SOX_OPTIONS"
+	    SOX_OPTIONS="$SOX_OPTIONS vol $3"
         fi
-
-	SOX_OPTIONS="$SOX_OPTIONS -"
-
-	if [ $SAMPRATE_OUT -ne $SAMPRATE_IN ]; then
-	    SOX_OPTIONS="$SOX_OPTIONS resample"
-	fi
     
         if [ $CHANS_OUT = 1 ]; then
 	    MODE="m"
@@ -222,7 +211,7 @@
 # TODO: extract the PEAK_LEVEL during the convertion
 SCALE=1
 if [ $NORMAL_LEVEL != 0 ]; then
-    PEAK_LEVEL=`sox $WORK -t wav /dev/null stat -v 2>&1 | grep -v ^sox` 
+    PEAK_LEVEL=`sox $WORK -n stat -v 2>&1 | grep -v ^sox` 
     # grep avoid sox warning messages
     
     SCALE=`echo "$NORMAL_LEVEL * $PEAK_LEVEL" | bc -l`
diff -urN rivendell-1.2.2/scripts/rd_rip_cd rivendell-1.2.2.gentoo/scripts/rd_rip_cd
--- rivendell-1.2.2/scripts/rd_rip_cd	2007-02-14 15:59:12.000000000 -0600
+++ rivendell-1.2.2.gentoo/scripts/rd_rip_cd	2009-03-18 11:14:57.000000000 -0500
@@ -69,14 +69,6 @@
     PARANOIA_FLAG="-P 10"
 fi
 
-#
-# Set Resample Algorithm
-#
-if [ $SAMPRATE -ne 44100 ]; then
-    RESAMPLE_FLAG="resample"
-#    RESAMPLE_FLAG="resample -q"
-fi
-
 # ############################################################################
 # Encode to PCM16
 if [ $FORMAT = 0 ]; then
@@ -84,7 +76,7 @@
 # ############################################################################
 # # No Normalization, do single pass
 #	echo "SINGLE PASS, PCM, NO NORMALIZATION!"
-	cdda2wav --quiet -D $RIPDEV -t $TRACK $PARANOIA_FLAG - | sox -t wav - -t raw -w -s -r $SAMPRATE -c $CHANS - $RESAMPLE_FLAG | rdfilewrite --channels=$CHANS --sample-rate=$SAMPRATE $WAVE
+	cdda2wav --quiet -D $RIPDEV -t $TRACK $PARANOIA_FLAG - | sox -t wav - -2 -r $SAMPRATE -c $CHANS -t raw - | rdfilewrite --channels=$CHANS --sample-rate=$SAMPRATE $WAVE
 # ############################################################################
 
 # ############################################################################
@@ -106,7 +98,7 @@
 # Phase Two - Encode audio to final destination
 #
 #	echo "PASS TWO, PCM, NORMALIZATION!"
-	sox -v $RATIO $WORK -t raw -w -s -c $CHANS -r $SAMPRATE - $RESAMPLE_FLAG | rdfilewrite --channels=$CHANS --sample-rate=$SAMPRATE $WAVE
+	sox $WORK -t raw -2 -c $CHANS -r $SAMPRATE - vol $RATIO | rdfilewrite --channels=$CHANS --sample-rate=$SAMPRATE $WAVE
     
 #
 # Clean Up
@@ -141,7 +133,7 @@
 # ############################################################################
 # # No Normalization, do single pass
 #	echo "SINGLE PASS, MPEG, NO NORMALIZATION!"
-	cdda2wav --quiet -D $RIPDEV -t $TRACK $PARANOIA_FLAG - | sox -t wav - -t raw -r $SAMPRATE -c $CHANS - $RESAMPLE_FLAG | toolame -t 0 -W -m $MODE -b $BITRATE -s $LAMERATE - $WAVE
+	cdda2wav --quiet -D $RIPDEV -t $TRACK $PARANOIA_FLAG - | sox -t wav - -r $SAMPRATE -c $CHANS -t raw - | toolame -t 0 -W -m $MODE -b $BITRATE -s $LAMERATE - $WAVE
 # ############################################################################
 
 # ############################################################################
@@ -163,7 +155,7 @@
 # Phase Two - Encode audio to final destination
 #
 #	echo "PASS TWO, MPEG, NORMALIZATION!"
-	sox -v $RATIO $WORK -t raw -c $CHANS -r $SAMPRATE - $RESAMPLE_FLAG | toolame -t 0 -W -m $MODE -b $BITRATE -s $LAMERATE - $WAVE
+	sox $WORK -c $CHANS -r $SAMPRATE -t raw - vol $RATIO | toolame -t 0 -W -m $MODE -b $BITRATE -s $LAMERATE - $WAVE
     
 #
 # Clean Up
